liferay-lima/liferay-portal
# Lps 202517 AI Generate images - add forms and mockup implementation

**Issues:**
https://liferay.atlassian.net/browse/LPS-202516 Add the new form to get Images from Dall-e
https://liferay.atlassian.net/browse/LPS-202517 Create the backend logic to call the client (by now the mock client with testing porpouses)
https://liferay.atlassian.net/browse/LPS-202518 Add the generated image selection screen


**Motivation**
This pull allows to generate from 1 to 4 images via Dalle API. Tests have been done using a mock client.

In future pulls we will save the selected images, and format them via selectable cards.

**How to test it:**
See https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2413429012/How+to+use+the+OpenAI+mock+client for reference.

1. Enable mock Client via Gogo shell (`scr:enable com.liferay.ai.creator.openai.web.internal.client.MockAICreatorOpenAIClient)`
2. Set VALID_API_KEY as Apy key for AI Creator in Instance Settings
3. Go to D&M and click Create AI Image (from the + dropdown)
4. On Description, add USER_IMAGES_URL_ + an url that exists. i.e `USER_IMAGES_URL_https://fastly.picsum.photos/id/1/5000/3333.jpg?hmac=Asv2DU3rA_5D1xSe22xZK47WEAN0wjWeFOhzd13ujW4`
<img width="1011" alt="AIScreeshot" src="https://github.com/liferay-lima/liferay-portal/assets/2728470/961e836b-a2df-43ea-8117-4e9c80455280">
